core/src/main/scala/com/banno/kafka/producer/ProducerImpl.scala

```diff
@@ -70,6 +72,21 @@ case class ProducerImpl[F[_], K, V](p: Producer[K, V])(implicit F: Async[F])
     Some(F.delay(jFuture.cancel(false)).void)
   }
 
+  /** The outer effect sends the record on a blocking context and is cancelable.
+    * The inner effect cancels the underlying send.
+    */
+  private def sendRaw2(
+      record: ProducerRecord[K, V],
+      callback: Try[RecordMetadata] => Unit,
```

Comment:
I'm curious - why the Try instead of Either? Is the thinking that we might end up naming this `sendRaw` in the end, so we'd need a different signature anyway?

Reply:
The main reason for `Try` here is that it aligns with `promise.complete` below.